src/core/server/saved_objects/migrations/utils.ts

```diff
@@ -16,6 +16,16 @@ import {
 
 /**
  * Merges two saved object migration maps.
+ *
+ * If there is a migration for a given version on only one of the maps,
+ * that migration function will be used:
+ *
+ * mergeSavedObjectMigrationMaps({ '1.2.3': f }, { '4.5.6': g }) -> { '1.2.3': f, '4.5.6': g }
+ *
+ * If there is a migration for a given version on both maps, the migrations will be composed:
+ *
+ * mergeSavedObjectMigrationMaps({ '1.2.3': f }, { '1.2.3': g }) -> { '1.2.3': (doc, context) => f(g(doc, context), context) }
+ *
```

Comment:
is this different than `mergeMigrationFunctionMaps` from kibana_utils ? what's the difference ? could we get rid of the one in utils and replace all usages with this one ?